content/crutchcorn/collections/art-of-accessibility/posts/art-of-a11y-semantic-markup/index.md

```diff
@@ -0,0 +1,486 @@
+    >
+      <code>console.log("Hello, world!");</code>
+    </div>
+    <div
+      role="tabpanel"
+      id="python-panel"
+      aria-labelledby="python-tab"
+      [hidden]="activeTab() !== 'python'"
+    >
+      <code>print("Hello, world!")</code>
+    </div>
+  </div>
+  `,
+})
+export class AppComponent {
+  activeTab = signal('javascript');
+
+  setActiveTab(val: string) {
+    this.activeTab.set(val);
+  }
+}
+```
+
+## Vue
+
+```vue
+<script setup>
+  import { ref } from 'vue'
+
+  const activeTab = ref('javascript')
+
+  function setActiveTab(val) {
+    activeTab.value = val
+  }
+</script>
+
+<template>
+  <div>
+    <ul role="tablist">
+      <li
+        role="tab"
+        id="javascript-tab"
+        :aria-selected="activeTab === 'javascript'"
+        aria-controls="javascript-panel"
+        @click="setActiveTab('javascript')"
+      >
+        JavaScript
+      </li>
+      <li
+        role="tab"
+        id="python-tab"
+        :aria-selected="activeTab === 'python'"
+        aria-controls="python-panel"
+        @click="setActiveTab('python')"
+      >
+        Python
+      </li>
+    </ul>
+    <div role="tabpanel" id="javascript-panel" aria-labelledby="javascript-tab" :hidden="activeTab !== 'javascript'">
+      <code>console.log("Hello, world!");</code>
+    </div>
+    <div role="tabpanel" id="python-panel" aria-labelledby="python-tab" :hidden="activeTab !== 'python'">
+      <code>print("Hello, world!")</code>
+    </div>
+  </div>
+</template>
+```
+
+<!-- ::end:tabs -->
+
+
+
+> 🎉 Tad-whoa. 😵‍💫
+
+![Two bullet points of the languages with a console.log below](./unstyled_tabs.png)
+
+> Are we sure this worked?
+
+Well, it's not the prettiest UI visually, but we can verify it's functionality by clicking on the `JavaScript` or `Python` text in order to show the `console.log` or `print` statements respectively.
+
+Now all we need to do is add a bit of CSS...
+
+<details>
+<summary>The required CSS for the tab component</summary>
+
+```css
+/* index.css */
+[role="tablist"] {
+  margin: 0;
+  padding: 0; 
+  display: flex;
+  gap: 0.25rem;
+}
+
+[role="tab"] {
+  display: inline-block;
+  padding: 1rem;
+  border: solid black;
+  border-width: 2px 2px 0 2px;
+  border-radius: 1rem 1rem 0 0;
+}
+
+[role="tab"]:hover {
+  background: #d3d3d3;
+}
+
+[role="tab"]:active {
+  background: #878787;
+}
+
+[role="tab"][aria-selected="true"] {
+  background: black;
+  color: white;
+}
+
+[role="tabpanel"] {
+  border: solid black;
+  border-width: 2px;
+  padding: 1rem;
+  border-radius: 0 1rem 1rem 1rem;
+}
+```
+
+</details>
+
+And tada! 🎉 (For real this time.)
+
+![The correct tab styling as outlined before](./styled_tabs.png)
+
+<!-- ::start:tabs -->
+
+## React
+
+<iframe data-frame-title="React Tab Component - StackBlitz" src="pfp-code:./art-of-a11y-react-tab-comp-1?embed=1&file=src/main.jsx" sandbox="allow-modals allow-forms allow-popups allow-scripts allow-same-origin"></iframe>
```

Comment:
Every iframe gives me an error saying "unable to run embedded project without proper isolation headers"
![image](https://github.com/user-attachments/assets/d8b53449-3630-4290-b328-795131312c48)

Reply:
This is a known bug unrelated to this branch